docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb

```diff
@@ -0,0 +1,524 @@
+    "\n",
+    "Since particles moving at large angles will not be seen in the synthetic radiograph, we will set an upper bound $\\theta_{max}$ on the argument here. This is equivalent to setting the `max_theta` keyword in `create_particles`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "arg = np.linspace(0, np.pi / 8, num=int(1e5))\n",
+    "prob = np.sin(arg) * np.exp(-(arg ** 2) / 0.1 ** 2)\n",
+    "prob *= 1 / np.sum(prob)\n",
+    "theta = np.random.choice(arg, size=int(nparticles), replace=True, p=prob)\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(6, 6))\n",
+    "theta_per_sa, bins = np.histogram(theta, bins=100, weights=1 / np.sin(theta))\n",
+    "ax.plot(bins[1:], theta_per_sa)\n",
+    "ax.set_xlabel(\"$\\\\theta$ (rad)\")\n",
+    "ax.set_ylabel(\"#/d$\\\\Omega$\")\n",
+    "ax.set_yscale(\"log\")\n",
+    "ax.set_xlim(0, np.pi / 2)\n",
+    "ax.set_ylim(None, 1e5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that the angular distributions are done, we will determine the energy (speed) for each particle. For this example, we will assume that the particle energy distribution is not a function of angle. We will create a Gaussian distribution of speeds with ~5% variance centered on a particle energy of 15 MeV."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v_cent = np.sqrt(2 * 15 * u.MeV / particle.mass).to(u.m / u.s).value\n",
+    "v0 = np.random.normal(loc=v_cent, scale=1e6, size=int(nparticles))\n",
+    "v0 *= u.m / u.s\n",
+    "\n",
+    "fig, ax = plt.subplots(figsize=(6, 6))\n",
+    "v_per_bin, bins = np.histogram(v0.si.value, bins=100)\n",
+    "ax.plot(bins[1:], v_per_bin)\n",
+    "ax.set_xlabel(\"v0 (m/s)\")\n",
+    "ax.set_ylabel(\"#\")\n",
+    "ax.axvline(x=1.05 * v_cent, label=\"+5%\", color=\"C1\")\n",
+    "ax.axvline(x=0.96 * v_cent, label=\"-5%\", color=\"C2\")\n",
+    "ax.legend()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we will construct velocity vectors centered around the z-axis for each particle."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vel = np.zeros([int(nparticles), 3]) * u.m / u.s\n",
+    "vel[:, 0] = v0 * np.sin(theta) * np.cos(phi)\n",
+    "vel[:, 1] = v0 * np.sin(theta) * np.sin(phi)\n",
+    "vel[:, 2] = v0 * np.cos(theta)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you want a velocity distribution centered on another vector, the function `proton_radiography._rot_a_to_b` can create an appropriate rotation matrix."
```

Comment:
If you're going to reference the `_rot_a_to_b` functionality, then it should not be private.   Either remove the reference from the notebook or refactor it to `rot_a_to_b` and properly document.

Reply:
I definitely agree, but it's such a general function that I don't think it belongs in `proton_radiography.py`. Where should we put it?

Reply:
🤔 It could possibly be put in `plasmapy.formulary.mathematics`.  I think that would make sense.

Reply:
Sounds good to me!